make image-test fails

```
+ make image-test
2	0+0 records in
3	0+0 records out
4	
5	Creating RAW device, this may take a moment...done
6	Using kernel parameter talos.platform=test
7	
8	Using kernel parameter talos.userdata=none
9	Using device /dev/loop1 as installation media
10	2019/08/16 20:03:32 partitioning /dev/loop1 - ESP
11	2019/08/16 20:03:32 partitioning /dev/loop1 - EPHEMERAL
12	2019/08/16 20:03:32 formatting partition /dev/loop1p1 - ESP as fat
13	2019/08/16 20:03:32 formatting partition /dev/loop1p2 - EPHEMERAL as xfs
14	
15	2019/08/16 20:03:32 WARNING: no ESP partition was found
16	2019/08/16 20:03:32 copying /usr/install/vmlinuz to /boot/default/vmlinuz
17	2019/08/16 20:03:32 copying /usr/install/initramfs.xz to /boot/default/initramfs.xz
18	
19	2019/08/16 20:03:32 writing /boot/syslinux/syslinux.cfg to disk
20	2019/08/16 20:03:32 writing syslinux label default to disk
21	2019/08/16 20:03:32 writing /boot/EFI/syslinux/syslinux.cfg to disk
22	2019/08/16 20:03:32 writing syslinux label default to disk
23	2019/08/16 20:03:32 failed to install extlinux: exit status 1
24	
25	make: *** [Makefile:204: image-test] Error 1
```
https://ci.dev.talos-systems.io/talos-systems/talos/2668/1/15

Added some logging to `pkg/blockdevice/probe/probe.go` and found `FileSystem()` was failing to open the device file:

```
+ make image-test
2    0+0 records in
3    0+0 records out
4    Creating RAW device, this may take a moment...done
5    Using kernel parameter talos.platform=test
6    Using kernel parameter talos.userdata=none
7    Using device /dev/loop1 as installation media
8    2019/09/09 19:55:45 failed to source provided userdata; falling back to defaults
9    2019/09/09 19:55:45 partitioning /dev/loop1 - ESP
10    2019/09/09 19:55:45 partitioning /dev/loop1 - EPHEMERAL
11    2019/09/09 19:55:45 formatting partition /dev/loop1p1 - ESP as fat
12    2019/09/09 19:55:45 formatting partition /dev/loop1p2 - EPHEMERAL as xfs
13    2019/09/09 19:55:45 filesystem.openfile open /dev/loop1p1: no such file or directory
14    2019/09/09 19:55:45 filesystem.openfile open /dev/loop1p2: no such file or directory
15    2019/09/09 19:55:45 devforfilesystemlabel.filterbylabel: no device found with label EPHEMERAL
16    github.com/talos-systems/talos/pkg/blockdevice/probe.filterByLabel
17    /src/pkg/blockdevice/probe/probe.go:221
18    github.com/talos-systems/talos/pkg/blockdevice/probe.DevForFileSystemLabel
19    /src/pkg/blockdevice/probe/probe.go:112
20    github.com/talos-systems/talos/internal/pkg/mount/manager/owned.MountPointsForDevice
21    /src/internal/pkg/mount/manager/owned/owned.go:34
22    github.com/talos-systems/talos/internal/pkg/installer.(*Installer).Install
23    /src/internal/pkg/installer/installer.go:75
24    github.com/talos-systems/talos/cmd/osctl/cmd.glob..func18
25    /src/cmd/osctl/cmd/install_linux.go:72
26    github.com/spf13/cobra.(*Command).execute
27    /go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:766
28    github.com/spf13/cobra.(*Command).ExecuteC
29    /go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:850
30    github.com/spf13/cobra.(*Command).Execute
31    /go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:800
32    github.com/talos-systems/talos/cmd/osctl/cmd.Execute
33    /src/cmd/osctl/cmd/root.go:96
34    main.main
35    /src/cmd/osctl/main.go:16
36    runtime.main
37    /toolchain/go/src/runtime/proc.go:200
38    runtime.goexit
39    /toolchain/go/src/runtime/asm_amd64.s:1337
40    2019/09/09 19:55:45 probe device for filesystem EPHEMERAL: no device found with label EPHEMERAL
41    make: *** [Makefile:224: image-test] Error 1
```

--- a/pkg/blockdevice/probe/probe.go
+++ b/pkg/blockdevice/probe/probe.go
@@ -13,6 +13,7 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+	"time"
 
 	"github.com/pkg/errors"
 	"github.com/talos-systems/talos/pkg/blockdevice"
@@ -55,9 +56,20 @@ func All() (all []*ProbedBlockDevice, err error) {
 // FileSystem probes the provided path's file system.
 func FileSystem(path string) (sb filesystem.SuperBlocker, err error) {
 	var f *os.File
-	if f, err = os.OpenFile(path, os.O_RDONLY|unix.O_CLOEXEC, os.ModeDevice); err != nil {
-		return nil, err
+	// Sleep for up to 5s to wait for kernel to create the necessary device files.
+	// If we dont sleep this becomes racy in that the device file does not exist
+	// and it will fail to open.
+	for i := 0; i <= 100; i++ {
+		if f, err = os.OpenFile(path, os.O_RDONLY|unix.O_CLOEXEC, os.ModeDevice); err != nil {
+			if os.IsNotExist(err) {
+				time.Sleep(50 * time.Millisecond)
+				continue
+			}
+			return nil, err
+		}
+		break
 	}
+
 	// nolint: errcheck
 	defer f.Close()
 

